pyi.py

```diff
@@ -1987,6 +1998,10 @@ def check_self_typevars(self, node: ast.FunctionDef | ast.AsyncFunctionDef) -> N
                 return_annotation=return_annotation,
             )
 
+    def check_arg_kinds(self, node: ast.FunctionDef | ast.AsyncFunctionDef) -> None:
+        for pos_or_kw in node.args.args[1:]:  # exclude "self"
+            self.error(pos_or_kw, Y062.format(arg=pos_or_kw.arg, method=node.name))
```

Comment:
This only allows protocols that use PEP-570 syntax, but typeshed doesn't use any PEP-570 syntax yet.

Also, what if it's a public protocol? In that case, it's probably a stub for a "real" protocol that exists at runtime, so stubtest will start complaining if a parameter is positional-only in the stub but positional-or-keyword at runtime. Maybe we should only complain about private protocols, similar to the way we only complain if a protocol/TypeVar/TypedDict is unused if the protocol/TypeVar/TypedDict is private?

Reply:
I fixed the code to also handle `__` (typeshed-primer was unhappy with me about that).

Not sure about public protocols; it's probably also a bug in the public code. Let's go over the typeshed hits first to see if there are real-world examples of this problem.

Reply:
We now handle both the old and new pos-only syntax. Since this discussion started, typeshed switched to PEP 570 and flake8-pyi started warning about the double underscore syntax, but for this check I think it's still useful to support both, in case some users are still using the old syntax in stubs.